### PR TITLE
Add the option to configure a PodDisruptionBudget

### PR DESCRIPTION
**Pull request by nweisenauer-sap**
**What this PR does / why we need it**:
Adds the option to configure a PodDisruptionBudget via the values.yaml to make the sa-issuer-discovery deployment more resilient.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature operator
Added the option to configure a PodDisruptionBudget via the helm chart values.
```

### CODE REVIEW

**Comment by ghost** [comment]
@nweisenauer-sap Thank you for your contribution.

**Comment by gardener-robot-ci-2** [comment]
Thank you @nweisenauer-sap for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: charts/service-account-issuer-discovery/templates/pdb.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+{{- if .Values.resilience.pdb.enabled }}
```
Comment (vpnachev) [inline charts/service-account-issuer-discovery/templates/pdb.yaml]: In gardener organization we are using the desired replicas as a switch to enable or not the pod disruption budget. See this [example](https://github.com/gardener/gardener/blob/master/charts/gardener/gardenlet/templates/poddisruptionbudget.yaml), please apply the same approach.

**Review by vpnachev — CHANGES_REQUESTED** [review]

File: charts/service-account-issuer-discovery/templates/pdb.yaml (hunk 1 above)
Comment (nweisenauer-sap) [reply charts/service-account-issuer-discovery/templates/pdb.yaml]: Thanks, I also added the option to configure the `unhealthyPodEvictionPolicy`, enabled by default with `AlwaysAllow`.

File: charts/service-account-issuer-discovery/templates/pdb.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+{{- if gt (int .Values.replicaCount) 1 }}
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: {{ include "name" . | quote }}
```
Comment (vpnachev) [inline charts/service-account-issuer-discovery/templates/pdb.yaml]:
```suggestion
  name: {{ include "name" . }}
```

File: charts/service-account-issuer-discovery/templates/pdb.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  namespace: {{ .Release.Namespace | quote }}
```
Comment (vpnachev) [inline charts/service-account-issuer-discovery/templates/pdb.yaml]:
```suggestion
  namespace: {{ .Release.Namespace }}
```

File: charts/service-account-issuer-discovery/templates/pdb.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+spec:
```
Comment (vpnachev) [inline charts/service-account-issuer-discovery/templates/pdb.yaml:15]:
```suggestion
  labels:
    app.kubernetes.io/name: {{ include "name" . }}
    app.kubernetes.io/instance: {{ .Release.Name }}
    app.kubernetes.io/managed-by: {{ .Release.Service }}
spec:
```

File: charts/service-account-issuer-discovery/templates/pdb.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+{{- if not (kindIs "invalid" .Values.resilience.pdb.maxUnavailable) }}
+    maxUnavailable: {{ toYaml .Values.resilience.pdb.maxUnavailable }}
+{{- end }}
```
Comment (vpnachev) [inline charts/service-account-issuer-discovery/templates/pdb.yaml]:
```suggestion
  maxUnavailable: 1
```

File: charts/service-account-issuer-discovery/templates/pdb.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+{{- if not (kindIs "invalid" .Values.resilience.pdb.minAvailable) }}
+    minAvailable: {{ toYaml .Values.resilience.pdb.minAvailable }}
+{{- end }}
```
Comment (vpnachev) [inline charts/service-account-issuer-discovery/templates/pdb.yaml]:
```suggestion
```

File: charts/service-account-issuer-discovery/templates/pdb.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+{{- if .Values.resilience.pdb.unhealthyPodEvictionPolicy }}
+    unhealthyPodEvictionPolicy: {{ .Values.resilience.pdb.unhealthyPodEvictionPolicy | quote }}
```
Comment (vpnachev) [inline charts/service-account-issuer-discovery/templates/pdb.yaml]:
```suggestion
{{- if semverCompare ">= 1.26-0" .Capabilities.KubeVersion.Version }}
  unhealthyPodEvictionPolicy: AlwaysAllow
```

File: charts/service-account-issuer-discovery/templates/pdb.yaml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+{{- end }}
+    selector:
+      matchLabels:
+          app.kubernetes.io/name: {{ include "name" . | quote }}
```
Comment (vpnachev) [inline charts/service-account-issuer-discovery/templates/pdb.yaml]:
```suggestion
  selector:
    matchLabels:
      app.kubernetes.io/name: {{ include "name" . }}
      app.kubernetes.io/instance: {{ .Release.Name }}
```

**Review by vpnachev — CHANGES_REQUESTED** [review]

File: charts/service-account-issuer-discovery/values.yaml [hunk 9]
```diff
@@ -49,6 +49,12 @@ autoscaling:
     maxReplicas: 4
     cpuTargetAverageUtilization: 80
 
+resilience:
+  pdb:
+    maxUnavailable: 1
+    # minAvailable: "50%"
+    unhealthyPodEvictionPolicy: AlwaysAllow
+
```
Comment (vpnachev) [inline charts/service-account-issuer-discovery/values.yaml]:
```suggestion
```

**Review by vpnachev — CHANGES_REQUESTED** [review]

**Review by vpnachev — APPROVED** [review]
/lgtm

**Pull request merged**